papas/simulator.py

```diff
@@ -293,8 +293,8 @@ def simulate(self, ptcs, history):
                 # to avoid numerical problems in propagation (and avoid making a particle that is not used)
                 continue
             pdebugger.info(str('Simulating {}'.format(ptc)))
-            # ptc = pfsimparticle(gen_ptc, len(self.simulated_particles))
-            self.history[ptc.uniqueid] = Node(ptc.uniqueid)
+            uniqueid = ptc.uniqueid()
```

Comment:
is uniqueid() doing anything complicated? 
it should only return _uniqueid
I would avoid declaring variables in this case, and would use ptc.uniqueid() directly here, 
so that we can directly see which uniqueid it is.